wodles/aws/tests/test_sqs_queue.py

```diff
@@ -0,0 +1,137 @@
+                                              iam_role_duration=kwargs["iam_role_duration"],
+                                              profile=None)
+
+    assert integration.sqs_name == kwargs['name']
+    assert integration.iam_role_arn == kwargs['iam_role_arn']
+    mock_get_sqs_url.assert_called_once()
+    mock_bucket_init.assert_called_once()
+    mock_sts_client.assert_called_with(access_key=None, secret_key=None)
+    mock_client.get_caller_identity.assert_called_once()
+
+
+@patch('sqs_queue.AWSSQSQueue._get_sqs_url', return_value=SAMPLE_URL)
+@patch('wazuh_integration.WazuhIntegration.get_sts_client')
+@patch('wazuh_integration.WazuhIntegration.__init__', side_effect=wazuh_integration.WazuhIntegration.__init__)
+def test_aws_sqs_queue_delete_message(mock_wazuh_integration, mock_sts_client, mock_get_url):
+    """Test 'delete_message' method sends the given message to SQS."""
+    instance = utils.get_mocked_aws_sqs_queue()
+    instance.delete_message(SAMPLE_MESSAGE)
+
+    instance.client.delete_message.assert_called_with(QueueUrl=SAMPLE_URL, ReceiptHandle=SAMPLE_MESSAGE["handle"])
+
+
+@patch('wazuh_integration.WazuhIntegration.get_sts_client')
+@patch('wazuh_integration.WazuhIntegration.__init__', side_effect=wazuh_integration.WazuhIntegration.__init__)
+def test_aws_sqs_queue_delete_message_handles_exception_when_deleting_message(mock_wazuh_integration, mock_sts_client):
+    """Test 'delete_message' handles exceptions raised when trying to delete a message from SQS."""
+    instance = utils.get_mocked_aws_sqs_queue()
+    instance.client.delete_message.side_effect = Exception
+
+    with pytest.raises(SystemExit) as e:
+        instance.delete_message(SAMPLE_MESSAGE)
+    assert e.value.code == utils.UNABLE_TO_FETCH_DELETE_FROM_QUEUE
+
+
+@patch('sqs_queue.AWSSQSQueue._get_sqs_url', return_value=SAMPLE_URL)
+@patch('wazuh_integration.WazuhIntegration.get_sts_client')
+@patch('wazuh_integration.WazuhIntegration.__init__', side_effect=wazuh_integration.WazuhIntegration.__init__)
+def test_aws_sqs_queue_fetch_messages(mock_wazuh_integration, mock_sts_client, mock_get_url):
+    """Test 'fetch_messages' method retrieves one or more messages from the specified queue."""
+    instance = utils.get_mocked_aws_sqs_queue()
+    mock_receive = instance.client.receive_message
+    mock_receive.return_value = SAMPLE_RAW_MESSAGE
+
+    raw_messages = instance.fetch_messages()
+    mock_receive.assert_called_with(QueueUrl=instance.sqs_url, AttributeNames=['All'],
+                                    MaxNumberOfMessages=10, MessageAttributeNames=['All'],
+                                    WaitTimeSeconds=20)
+    assert type(raw_messages) is dict
+
+
+@patch('wazuh_integration.WazuhIntegration.get_sts_client')
+@patch('wazuh_integration.WazuhIntegration.__init__', side_effect=wazuh_integration.WazuhIntegration.__init__)
+def test_aws_sqs_queue_fetch_messages_handles_exception_when_getting_messages(mock_wazuh_integration, mock_sts_client):
```

Comment:
Just as a side note, no need to change anything: usually, the tests in the framework simply add `_ko` to the test name when it tests whether exceptions are being correctly handled. For example:
https://github.com/wazuh/wazuh/blob/9300dac1f43fb1245a7363dd528fc05b9188d350/framework/wazuh/core/tests/test_utils.py#L192-L193
and
https://github.com/wazuh/wazuh/blob/9300dac1f43fb1245a7363dd528fc05b9188d350/framework/wazuh/core/tests/test_utils.py#L206-L207